### PR TITLE
feat(lhls): add lhls support

### PR DESCRIPTION
**Pull request by kongjianneu**

### CODE REVIEW

File: framework/demuxer/demuxer_service.cpp [hunk 1]
```diff
@@ -287,6 +287,15 @@ namespace Cicada {
         return mDemuxerPtr->GetRemainSegmentCount(index);
     }
 
+    bool demuxer_service::isRealTimeStream(int index)
+    {
+        if (nullptr == mDemuxerPtr) {
+            return false;
+        }
+
+        return (!mDemuxerPtr->isPlayList()) || mDemuxerPtr->isRealTimeStream(index);
```
Comment (pingkai) [inline framework/demuxer/demuxer_service.cpp]: 1. If not a playList demuxer, then it is a realtime stream ?
2. Is the parameter index usefull?

File: framework/demuxer/play_list/HLSStream.cpp [hunk 2]
```diff
@@ -119,10 +129,32 @@ namespace Cicada {
         } else {
             ret = mSegDecrypter->Read(const_cast<uint8_t *>(buffer), size);
         }
-
+
```
Comment (pingkai) [inline framework/demuxer/play_list/HLSStream.cpp]: use git clang format to format the code

File: framework/demuxer/play_list/HLSStream.cpp [hunk 3]
```diff
@@ -95,6 +96,15 @@ namespace Cicada {
 
         ret = pHandle->readSegment(buffer, size);
 
+        if (ret == 0) {
+            int move_ret = pHandle->moveToNextPartialSegment();
+            if (move_ret == 1) {
```
Comment (pingkai) [inline framework/demuxer/play_list/HLSStream.cpp]: What the mean of 1? Use a enum or bool?

**Review by pingkai — CHANGES_REQUESTED** [review]

File: framework/demuxer/play_list/HLSStream.cpp (hunk 3 above)
Comment (kongjianneu) [reply framework/demuxer/play_list/HLSStream.cpp]: modified

File: framework/demuxer/play_list/HLSStream.cpp (hunk 2 above)
Comment (kongjianneu) [reply framework/demuxer/play_list/HLSStream.cpp]: modified

**Review by pingkai — APPROVED** [review]

**Pull request merged**